src/test/java/com/fasterxml/jackson/dataformat/xml/XmlMapperTest.java

```diff
@@ -1,25 +1,19 @@
 package com.fasterxml.jackson.dataformat.xml;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
 
-import org.junit.Rule;
-import org.junit.Test;
+import org.junit.jupiter.api.Test;
 import org.jvnet.hudson.test.JenkinsRule;
-import org.jvnet.hudson.test.RealJenkinsRule;
+import org.jvnet.hudson.test.junit.jupiter.WithJenkins;
 
 import java.nio.charset.StandardCharsets;
 
-public class XmlMapperTest {
-
-    @Rule public RealJenkinsRule rr = new RealJenkinsRule();
+@WithJenkins
```

Comment:
this is not the same thing as `RealJenkinsRule`, and the test is using this for good reason (JenkinsRule uses a non realistic classpath)

Reply:
Well in that case upgrading to `JUnit5JenkinsRule` or JUnit5 won't be possible I guess. 

I was unaware that his is required in order to _properly_ test this plugin, since other library plugins [also use JUnit 5](https://github.com/search?q=org%3Ajenkinsci+SmokeTest+jupiter+language%3AJava+&type=code). Would you care to elaborate why this is important so I can understand it better?

Reply:
> Well in that case upgrading to JUnit5JenkinsRule or JUnit5 won't be possible I guess.

at least until such time that an equivalent is introduced in `jenkins-test-harness`.

> I was unaware that this is required in order to properly test this plugin, since other library plugins [also use JUnit 5](https://github.com/search?q=org%3Ajenkinsci+SmokeTest+jupiter+language%3AJava+&type=code).

I would say that at least several of those smoke tests are close to useless (see below) :-)

> Would you care to elaborate why this is important so I can understand it better?

when running with `JenkinsRule`. `JenkinsSessionRule` or `WithJenkins` the resulting Jenkins instance is run in the same JVM as the test.  The test has its classpath set by maven-surefire, and as such is a big long flat list of jars.

In this scenario every class has the same classloader§, and as such all classess can see and load any other classes.

When you run Jenkins this is not the case at all as Jenkins has a different classloader for each plugin (and Jenkins itself) ([more details here](https://www.jenkins.io/doc/developer/plugin-development/dependencies-and-class-loading/)).

For a lot of tests this may not matter and there are some extra things in place to try and help developers stay inside the bubble where this should work (e.g. the [`requireUpperBounds` enforcer rule](https://github.com/jenkinsci/plugin-pom/blob/ed08c914961d17cd9fe6a1be9aa4e02742a9dfcc/pom.xml#L596-L651)) .

However, if a plugin is performing class loading, or is using special classloading then this becomes important, as the classloader they would inside Jenkins would be vastly different to the one when run by surefire.  
e.g. Anything using reflection (`Class.forName`) to load classes. 
in this case the `Class.forName(...)` can work in the flat hierachy but be completely broken when run in Jenkins.

There can be other subtleties where the difference shows, 

for example as noted by the original author of the test (#82)

> I included a new integration test with RealJenkinsRule that works against Jenkins 2.297 and earlier, as well as tip-of-trunk with https://github.com/jenkinsci/jenkins/pull/5604. Without https://github.com/jenkinsci/jenkins/pull/5604, the test fails on tip-of-trunk with the same error as in https://github.com/jenkinsci/azure-storage-plugin/issues/194. So somehow we end up relying on the old version of Woodstox (3.x) from Jenkins core, even though jackson-dataformat-xml depends on a recent version of Woodstox as a compile-time dependency. I am not sure why. The test passes with regular JenkinsRule, so this must be a classpath issue with jenkins.war. This would be a useful starting point for anyone who is investigating removing Woodstox from Jenkins core.



§ not quite, as the JDK classes use the system classloader and their may be some extra things, but by inlarge you can think of it that every class can see every other class.

Reply:
Thank you very much for taking the time to explain it further, much appreciated!

So unless we get a `JUnit5RealJenkinsRule` there is no way to make these kind of tests to validate properly with JUnit5. I already had a closer look into the topic a while ago, maybe it's about time to pick it up again.

Thanks again, I'll close the PR.